cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -76,15 +76,16 @@ public void setRepository(FullHttpRequest request, HttpResponder responder,
       }
 
       repoRequest.getConfig().validate();
+      String responseMessage;
       if (repoRequest.shouldTest()) {
-        // TODO: CDAP-20252, add the validate logic once the SourceControlManager module is ready
+        sourceControlService.testRepositoryConfig(namespace, repoRequest.getConfig());
+        responseMessage = "Successfully validated the repository configuration.";
+      } else {
+        sourceControlService.setRepository(namespace, repoRequest.getConfig());
+        responseMessage = String.format("Updated repository configuration for namespace '%s'.",
```

Comment:
We should return the updated config rather than a string

Reply:
updated